### PR TITLE
Fix symbols dropped in match statement

### LINKED ISSUE #28780: No suggestions for variables defined in match patterns of a match statement
**Description:**
$title.

**Steps to reproduce:**
```ballerina
public function main() {
    any v = {x1: 1, b: true};
    match v {
        var  {x1, b: x2} => {
            any|error a = x<cursor>
        }
    }
}
```
match_vars

```ballerina
public function main() {
    error v = error("err", code = 1001);
    match v {
        error(var x1, code = var x2) => {
            any|error a = x<cursor>
        }
    }
}
```

match_vars_2

**Affected Versions:**
slalpha2

### LINKED ISSUE #28980: Symbols are dropped within the match statement block
**Description:**
Consider the following example,

```
function bar (string | int | boolean i)  returns (string | int | boolean) {
    string | int | boolean var1 =  "Test";
    [string, int, boolean] var2 = ["Foo", 12, true];
    int var4 = 12;
    string var5 = "test";
    Person p = { name : "Tom", age : 10};

    match var4 {
        var t => {
            <cursor>
        }
    }

    return 1;
}

type Person record {
    string name;
    int age;
};

```

At the cursor position, symbol `t` is missing

### PR DESCRIPTION
**Pull request by dulajdilshan**
Fixes #28980 
Fixes #28780 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolLookupTest.java [hunk 1]
```diff
@@ -317,6 +317,31 @@ public void testSymbolLookupWithAnnotationOnFunction() {
...
+        assertList(symbolsInMappingMatch, Arrays.asList("a", "func", "v", "x1", "x2"));
+
+        Map<String, Symbol> symbolsInTupleMatch = getSymbolsInFile(model, srcFile, 25, 26, moduleID);
+        assertList(symbolsInTupleMatch, Arrays.asList("a", "func", "v", "x3", "x4"));
+
+        Map<String, Symbol> symbolsInErrorMatch = getSymbolsInFile(model, srcFile, 29, 26, moduleID);
+        assertList(symbolsInErrorMatch, Arrays.asList("a", "func", "v", "x5", "x6"));
+
+        Map<String, Symbol> symbolsInVarMatch = getSymbolsInFile(model, srcFile, 33, 26, moduleID);
+        assertList(symbolsInVarMatch, Arrays.asList("a", "func", "v", "x7"));
```
Comment (pubudu91) [inline tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolLookupTest.java:342]: Shall we add some cases where the position is within the match pattern etc.? Also, I think these can be fed in through a data provider

Comment (dulajdilshan) [reply tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolLookupTest.java:342]: Ack

Comment (dulajdilshan) [reply tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolLookupTest.java]: @pubudu91 Any suggestions to add test cases for scenarios within the pattern?

Comment (pubudu91) [reply tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolLookupTest.java]: Nothing in particular. What I meant was, all the current positions in the test seem to be positions within the block statement of the match clause. My suggestion was to add some cases where the positions are within the match pattern of the clause as well.

**Comment by pubudu91** [comment]
Shall we look in to addressing the comments received in Slack as well?

**Review by pubudu91 — APPROVED** [review]

**Pull request merged**